### ISSUE TITLE
Materials foundation: `AbstractDielectric` and `AbstractConductor`

### ISSUE THREAD

**Issue by gvcallen** [paramrf/paramrf#60]
Part of #59

`area: materials` · `size: L` · `breaking` · `depends: none`

### Context

Today `epr`, `tand` and `rho` are real scalars on each line, so dispersion is
structurally impossible — not because the formulations cannot express it, but
because the material enters as a number. Replacing them with sub-modules makes
every coefficient a `Param`, so a substrate model can be regressed against
measured S-parameters *through* a line, and one implementation serves coax,
microstrip, stripline and CPW forever.

This issue builds the material package standalone. It touches no line model, so
it can land and be tested in isolation.

### Interfaces

```python
class AbstractDielectric(Module):
    def epsilon_r(self, freq: Frequency) -> jnp.ndarray:
        """Total complex relative permittivity, shape (freq.npoints,).

        Convention: eps_r = eps' - 1j*eps'', with eps'' >= 0 for a passive
        medium. The returned value INCLUDES any static conductivity
        contribution, so callers never see the split.
        """

class AbstractConductor(Module):
    def surface_impedance(self, freq: Frequency) -> jnp.ndarray:
        """Complex surface impedance in ohm per square, shape (npoints,)."""

    def sigma(self, freq: Frequency) -> jnp.ndarray:
        """Bulk conductivity, for formulations that need it directly."""
```

### Dielectric implementations

- `ConstantDielectric(eps_r, tand=0.0, sigma=0.0)` — the default; reproduces
  today's behaviour when `sigma=0`.
- `DjordjevicSarkar(eps_r, tand, f_low=1e3, f_high=1e12, f_ref=1e9, sigma=0.0)` —
  causal wideband Debye. Port the maths from
  `skrf.media.mline.MLine.analyse_dielectric`.
- `MultipoleDebye(eps_inf, poles, sigma=0.0)` — N-pole, all coefficients fittable.
- `ColeCole(eps_inf, delta_eps, f_relax, alpha, sigma=0.0)`.
- `TabulatedDielectric(f, eps_r_values, method='linear')` — interpolated vendor or
  measured data.

### Conductor implementations

- `Bulk(rho)` — smooth bulk metal, `Zs = sqrt(1j*w*mu/sigma)`.
- `RoughConductor(rho, rms_roughness, model='hammerstad')` — Hammerstad correction
  factor, with `'huray'` as a follow-up.

Keep the field named `rho` to match current code; add a `Bulk.from_sigma()`
classmethod rather than overloading one field with two conventions.

### The conductivity term (CST-style lossy dielectric)

A CST "normal" material specifies electrical conductivity σ instead of, or
alongside, tanδ. Loss terms are additive in the imaginary part:

```
eps_r(w) = eps'(w) - 1j*( eps'(w)*tand + sigma/(w*eps_0) )
```

Keep σ as a distinct field rather than folding it into tanδ, because the two
scale oppositely: constant tanδ gives G proportional to ω, while constant σ gives
a **frequency-independent G**. Neither can express the other. Negligible for PCB
laminates, but it is the dominant loss mechanism for on-chip interconnect over
conductive silicon, and it matters for semi-insulating GaAs, lossy ground, and
biological media.

### Known interface leak — coaxial internal impedance

The ohm-per-square interface is geometry-free by design, which is right for
planar lines. Coax is the exception: the Schelkunoff solution needs the conductor
*radius* inside a Bessel function ratio, and Tesche's equivalent circuit needs the
internal inductance of the specific rod or tube.

Resolution: `surface_impedance` covers planar lines and the coax high-frequency
limit. Coax keeps its own `formulation` field selecting Tesche vs Schelkunoff, and
that formulation consumes `conductor.sigma(freq)` plus the radius directly.
Document this split rather than contorting the interface to hide it.

### Task

- New package, suggested `pmrf/materials/` with `dielectric.py` and
  `conductor.py`, re-exported at `pmrf.materials`. Confirm placement before
  building out.
- Every numeric field is a `Param` with an appropriate constraint:
  `GreaterThan(1.0)` for permittivity, `Positive()` for tanδ and σ.
- Converters so `dielectric=4.3` builds `ConstantDielectric(4.3)`,
  `dielectric=(4.3, 0.02)` builds `ConstantDielectric(4.3, 0.02)`, and
  `conductor=1.68e-8` builds `Bulk(rho=1.68e-8)`. Follow the existing `as_param`
  converter idiom in `physical.py`.
- Guard the σ term at DC using the double-`where` pattern so the gradient is
  finite too, not just the value.

### Acceptance criteria

- `ConstantDielectric(4.3, 0.02).epsilon_r(freq)` equals `4.3*(1 - 0.02j)`.
- `DjordjevicSarkar` matches scikit-rf's `analyse_dielectric` to 1e-10 relative
  over 1 kHz – 100 GHz.
- A σ-only dielectric produces a frequency-independent G once used in a line.
- `Bulk` reproduces the existing `R_skin` terms in `TescheCoaxialSolver` and
  Wheeler's `R = (1/W)*sqrt(2*mu_0*rho*w)` to float tolerance.
- `RoughConductor(rms_roughness=0)` is identical to `Bulk`.
- `jax.grad` through every material model is finite, including at band edges.
- Each model round-trips through `pmrf.serialization`.

### Files

- `pmrf/materials/` (new)
- `pmrf/__init__.py`, `pmrf/models/__init__.py` re-exports
- `tests/test_materials/` (new)

---
